src/Console/InfectionContainer.php

```diff
@@ -170,10 +174,47 @@ public function __construct(array $values = [])
 
             return (new MutatorsGenerator($mutatorConfig))->generate();
         };
+
+        $this['environment.debugger.disabled'] = function (): bool {
+            return '' == trim((string) getenv(XdebugHandler::ENV_DISABLE_XDEBUG));
+        };
+
+        $this['environment.xdebug.loaded'] = function (): bool {
+            return \extension_loaded('xdebug');
+        };
+
+        $this['environment.phpdbg.used'] = function (): bool {
+            return PHP_SAPI == 'phpdbg';
+        };
+    }
+
+    public function setInfectionConfigInitializer(\Closure $initializer)
+    {
+        $this[self::KEY_INFECTION_CONFIG] = $initializer;
+    }
+
+    public function getInfectionConfig(): InfectionConfig
```

Comment:
what is so special with infection config that it has its own method?

If we are implementing `Psr\Container\ContainerInterface`, I would stick to its public interface without any additional methods, preserving `Psr\Container\ContainerInterface` typehints rather than our `InfectionContainer`. Otherwise, why do we need to implement `Psr\Container\ContainerInterface`?